### PR TITLE
add log10_tags context manager and with_tags decorator

### PR DESCRIPTION
**Pull request by wenzhe-log10**

### CODE REVIEW

File: log10/load.py [hunk 1]
```diff
@@ -166,6 +169,64 @@ def last_completion_id(self):
         return response["completionID"]
 
 
+@contextmanager
+def add_tags(tags: list[str]):
```
Comment (wenzhe-log10) [inline log10/load.py:173]: naming, `log10_tags`?

File: log10/load.py [hunk 2]
```diff
@@ -166,6 +215,60 @@ def last_completion_id(self):
...
+    >>>         ],
+    >>>     )
+    >>>     print(completion.choices[0].message)
+    """
+    tags_manager = TagsManager(tags)
+    with tags_manager:
+        yield
+
+
+def with_tags(tags: list[str]):
```
[the remark this answers is not in the document]
Comment (wenzhe-log10) [reply log10/load.py]: added

File: log10/load.py [hunk 3]
```diff
@@ -113,40 +113,89 @@ def get_session_id():
 #
 session_id_var = contextvars.ContextVar("session_id", default=get_session_id())
 last_completion_response_var = contextvars.ContextVar("last_completion_response", default=None)
+# is this tags_var for log10_session?
+# first how does this tags_var get set and passed around? what's the option to set it by users
 tags_var = contextvars.ContextVar("tags", default=[])
+extra_tags_var = contextvars.ContextVar("extra_tags", default=[])
```
[the remark this answers is not in the document]
Comment (wenzhe-log10) [reply log10/load.py]: removed extra_tags_var

File: log10/load.py [hunk 4]
```diff
@@ -113,40 +113,89 @@ def get_session_id():
...
+    def __init__(self, tags: list[str] = None):
+        self.tags = self._validate_tags(tags) or []
+
+    @staticmethod
+    def _validate_tags(tags: list[str] | None) -> list[str]:
+        if tags is None:
+            return None
+
+        if not isinstance(tags, list):
+            logger.error("tags must be a list")
```
[the remark this answers is not in the document]
Comment (wenzhe-log10) [reply log10/load.py]: updated the message.

File: log10/load.py [hunk 5]
```diff
@@ -113,40 +113,89 @@ def get_session_id():
...
+            return None
+
+        validated_tags = []
+        for tag in tags:
+            if not isinstance(tag, str):
+                logger.warning(f"All tags must be strings, found {tag} of type {type(tag)}")
```
[the remark this answers is not in the document]
Comment (wenzhe-log10) [reply log10/load.py]: updated the message.

File: log10/load.py [hunk 6]
```diff
@@ -113,40 +113,89 @@ def get_session_id():
 #
 session_id_var = contextvars.ContextVar("session_id", default=get_session_id())
 last_completion_response_var = contextvars.ContextVar("last_completion_response", default=None)
+# is this tags_var for log10_session?
+# first how does this tags_var get set and passed around? what's the option to set it by users
```
[the remark this answers is not in the document]
Comment (wenzhe-log10) [reply log10/load.py]: removed

**Review by nqn — APPROVED** [review]
Since the change behavior of nested session tags - swing this by the broader team for input

**Pull request merged**